### PR TITLE
Added simple gate and buoy AI nodes

### PR DESCRIPTION
**Pull request by irwineffect**

### CODE REVIEW

File: src/ai/buoy_jirwin.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+class Node():
+    def __init__(self):
+        self.pub = rospy.Publisher('control', control, queue_size=1)
+        self.sub = rospy.Subscriber('vision/start_gate', vision_pos, self.callback)
+        self.state = "SEARCHING"
+
+    def callback(self, vision_result):
+        msg = control()
+        rospy.info("state: {}".format(self.state))
+        #check for empty message, we can't see a buoy
```
Comment (skallaher) [inline src/ai/buoy_jirwin.py:17]: This is subscribed to the gate topic, why does this comment mention buoys?

File: src/ai/gate_jirwin.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import rospy
+from robosub.msg import visionPos as vision_pos
+from robosub.msg import control
+
+
+class Node():
+    def __init__(self):
+        self.pub = rospy.Publisher('control', control, queue_size=1)
+        self.sub = rospy.Subscriber('vision/buoy', vision_pos, self.callback)
```
Comment (skallaher) [inline src/ai/gate_jirwin.py]: Your subscribers seem backwards. The buoy processing is subscribed to the gate topic and the gate processing to the buoy topic

**Review by skallaher — APPROVED** [review]
Double check your topic subscriptions. Other than that, looks good to me.

File: src/ai/buoy_jirwin.py (hunk 1 above)
Comment (irwineffect) [reply src/ai/buoy_jirwin.py:17]: fixed by d21bdb4

File: src/ai/gate_jirwin.py (hunk 2 above)
Comment (irwineffect) [reply src/ai/gate_jirwin.py]: fixed by d21bdb4

**Pull request merged**